tests/nc-client/main.c

```diff
@@ -16,11 +18,40 @@ struct senddata {
 	int conndone;
 	void * connect_cookie;
 	void * write_cookie;
+	void * read_cookie;
+	uint8_t shutdown;
```

Comment:
I'd call this something like `uint8_t dummybuf[1];` to indicate that it's (a) a buffer and (b) not intended to actually be used.  It's far from clear what a variable `uint8_t shutdown` should be used for.